### ISSUE TITLE
'disabling non-fixed-group key exchange algorithms because we cannot find moduli file' error

### ISSUE THREAD

**Issue by imTHAI** [thinkst/opencanary#375]
**Describe the bug**
Hello,

I noticed in my logs, each time someone tries to ssh the container, I have the same error message
```
[...]
024-10-13T07:01:42+0200 [opencanary.modules.ssh.HoneyPotSSHFactory#info] disabling non-fixed-group key exchange algorithms because we cannot find moduli file
{"dst_host": "10.4.3.15", "dst_port": 22, "local_time": "2024-10-13 05:01:42.114899", "local_time_adjusted": "2024-10-13 07:01:42.114904", "logdata": {"SESSION": "14"}, "logtype": 4000, "node_id": "opencanary-1", "src_host": "147.185.132.40", "src_port": 58316, "utc_time": "2024-10-13 05:01:42.114902"}
2024-10-13T07:01:52+0200 [opencanary.modules.ssh.HoneyPotTransport#info] connection lost
2024-10-13T07:37:12+0200 [opencanary.modules.ssh.HoneyPotSSHFactory#info] disabling non-fixed-group key exchange algorithms because we cannot find moduli file
{"dst_host": "10.4.3.15", "dst_port": 22, "local_time": "2024-10-13 05:37:12.475616", "local_time_adjusted": "2024-10-13 07:37:12.475622", "logdata": {"SESSION": "15"}, "logtype": 4000, "node_id": "opencanary-1", "src_host": "92.255.85.253", "src_port": 52827, "utc_time": "2024-10-13 05:37:12.475620"}
2024-10-13T07:37:12+0200 [opencanary.modules.ssh.HoneyPotTransport#info] connection lost
2024-10-13T07:49:57+0200 [opencanary.modules.ssh.HoneyPotSSHFactory#info] disabling non-fixed-group key exchange algorithms because we cannot find moduli file
{"dst_host": "10.4.3.15", "dst_port": 22, "local_time": "2024-10-13 05:49:57.098087", "local_time_adjusted": "2024-10-13 07:49:57.098101", "logdata": {"SESSION": "16"}, "logtype": 4000, "node_id": "opencanary-1", "src_host": "92.255.85.107", "src_port": 42200, "utc_time": "2024-10-13 05:49:57.098097"}
2024-10-13T07:49:57+0200 [opencanary.modules.ssh.HoneyPotTransport#info] connection lost
2024-10-13T08:18:26+0200 [opencanary.modules.ssh.HoneyPotSSHFactory#info] disabling non-fixed-group key exchange algorithms because we cannot find moduli file
{"dst_host": "10.4.3.15", "dst_port": 22, "local_time": "2024-10-13 06:18:26.050742", "local_time_adjusted": "2024-10-13 08:18:26.050749", "logdata": {"SESSION": "17"}, "logtype": 4000, "node_id": "opencanary-1", "src_host": "45.79.181.251", "src_port": 50714, "utc_time": "2024-10-13 06:18:26.050746"}
2024-10-13T08:18:26+0200 [opencanary.modules.ssh.HoneyPotTransport#debug] kex alg=b'curve25519-sha256' key alg=b'rsa-sha2-512'
2024-10-13T08:18:26+0200 [opencanary.modules.ssh.HoneyPotTransport#debug] outgoing: b'aes128-ctr' b'hmac-sha2-256' b'none'
2024-10-13T08:18:26+0200 [opencanary.modules.ssh.HoneyPotTransport#debug] incoming: b'aes128-ctr' b'hmac-sha2-256' b'none'
2024-10-13T08:18:27+0200 [opencanary.modules.ssh.HoneyPotTransport#debug] NEW KEYS
2024-10-13T08:18:27+0200 [opencanary.modules.ssh.HoneyPotTransport#info] connection lost
2024-10-13T08:18:27+0200 [opencanary.modules.ssh.HoneyPotSSHFactory#info] disabling non-fixed-group key exchange algorithms because we cannot find moduli file
{"dst_host": "10.4.3.15", "dst_port": 22, "local_time": "2024-10-13 06:18:27.845241", "local_time_adjusted": "2024-10-13 08:18:27.845248", "logdata": {"SESSION": "18"}, "logtype": 4000, "node_id": "opencanary-1", "src_host": "45.79.181.251", "src_port": 50718, "utc_time": "2024-10-13 06:18:27.845246"}
2024-10-13T08:18:28+0200 [opencanary.modules.ssh.HoneyPotTransport#info] Disconnecting with error, code 3
        reason: b"couldn't match all kex parts"
[...]
```

As you can see at the end of my extract, I have sometimes this error:
```
 2024-10-13T08:18:28+0200 [opencanary.modules.ssh.HoneyPotTransport#info] Disconnecting with error, code 3
        reason: b"couldn't match all kex parts"
```
This line even appears in red in the logs !

But I don't know if both errors are related.
I'm not sure it's a bug. I must have forgotten a parameter. 

I've installed opencanary simply using the Dockerfile and the proposed [opencanary.conf](https://github.com/thinkst/opencanary/blob/master/data/.opencanary.conf) config file

**Comment by thinkst-az** [comment]
Hi @imTHAI ,

Sorry you had to hit this. Could you check if these 2 files exists in the container by running this command?

```
ls -al /etc/ssh/moduli /private/etc/moduli
```

The current implementation relies on the SSH moduli already being generated in those 2 files (typically by sshd) on the host. It's quite possible this base container image doesn't have that.

Do you still get an SSH alert, even with this error?

And lastly,  could you send us the verbose output from the SSH client with `ssh -vv <IP>`?

**Comment by thinkst-az** [comment]
Hi @imTHAI , we pushed a fix for the missing moduli issue in v0.9.5. It might not be the fix for this issue you've hit though.

Could you please update and retry ssh with `ssh -vv user@<IP>` and send us the verbose logs to debug further?

**Comment by imTHAI** [comment]
@thinkst-az hello, I don't get these error message anymore, it's fixed. Thank you
